src/functions/reportGen.ts

```diff
@@ -23,33 +22,27 @@ const reportGen: Handler = async (event: any, context?: Context, callback?: Call
   }
   const lambdaService = new LambdaService(new LambdaClient({}));
   const reportService: ReportGenerationService = new ReportGenerationService(new TestResultsService(lambdaService), new ActivitiesService(lambdaService));
-  const atfReportPromises: Promise<PutObjectRequest>[] = [];
-
   const sendATFReport: SendATFReport = new SendATFReport();
 
-  event.Records.forEach((record: any) => {
-    const recordBody = JSON.parse(record?.body)
-    const visit: any = unmarshall(recordBody?.dynamodb.NewImage);
+  console.debug("Services injected, looping over sqs events");
+  try {
+    for (const record of event.Records) {
+      const recordBody = JSON.parse(record?.body);
+      const visit: any = unmarshall(recordBody?.dynamodb.NewImage);
 
-    if (visit) {
-      const atfReportPromise = reportService
-        .generateATFReport(visit)
-        .then((generationServiceResponse) => {
-          return sendATFReport.sendATFReport(generationServiceResponse, visit);
-        })
-        .catch((error: any) => {
-          console.log(error);
-          throw error;
-        });
+      console.debug(`visit is: ${JSON.stringify(visit)}`);
```

Comment:
Similar to Leo's comment with regards to the e-mail, we don't want to log the entire object here as it includes the e-mails etc.

But visit.id would be helpful.

Reply:
Although did just spot it's a debug log, so we are probably okay actually.